includes/executor/class-urlslab-executor-generate-url-context.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+use Urlslab_Vendor\GuzzleHttp;
+use Urlslab_Vendor\OpenAPI\Client\Configuration;
+use Urlslab_Vendor\OpenAPI\Client\Model\DomainDataRetrievalAugmentPrompt;
+use Urlslab_Vendor\OpenAPI\Client\Model\DomainDataRetrievalAugmentRequest;
+use Urlslab_Vendor\OpenAPI\Client\Model\DomainDataRetrievalComplexAugmentRequest;
+use Urlslab_Vendor\OpenAPI\Client\Urlslab\ContentApi;
+
+class Urlslab_Executor_Generate_Url_Context extends Urlslab_Executor {
+	const TYPE = 'generate-url-context';
+
+	protected function schedule_subtasks( Urlslab_Data_Task $task_row ): bool {
+		$data = $task_row->get_data();
+		if ( is_array( $data ) && ! empty( $data ) ) {
+			$urls     = $data['urls'];
+			$executor = new Urlslab_Executor_Download_Urls_Batch();
+			$executor->schedule( $urls, $task_row );
+		} else {
+			$this->execution_finished( $task_row );
+		}
+
+		return true;
+	}
+
+	protected function on_all_subtasks_done( Urlslab_Data_Task $task_row ): bool {
+		if ( isset( $task_row->get_data()['process_id'] ) ) {
+			// waiting for process res
+			try {
+				$rsp = Urlslab_Connection_Augment::get_instance()->get_process_result( $task_row->get_data()['process_id'] );
+
+				switch ( $rsp->getStatus() ) {
+					case 'SUCCESS':
+						$task_row->set_result( $rsp->getResponse()[0] );
+						$this->execution_finished( $task_row );
+						return true;
+					case 'ERROR':
+						$task_row->set_result( $rsp->getResponse()[0] );
+						$this->execution_failed( $task_row );
+						return true;
+					default: //pending
+						$this->execution_postponed( $task_row, 3 );
+
+						return false;
+				}
+			} catch ( \Urlslab_Vendor\OpenAPI\Client\ApiException $exception ) {
+				$this->execution_failed( $task_row );
+
+				return true;
+			}
+		}
+
+		$childs = $this->get_child_tasks( $task_row, Urlslab_Executor_Download_Urls_Batch::TYPE );
+		if ( empty( $childs ) ) {
+			$this->execution_failed( $task_row );
+
+			return false;
+		}
+
+		$batch_result     = self::get_executor( Urlslab_Executor_Download_Urls_Batch::TYPE )->get_task_result( $childs[0] );
+		$docs = array();
+		foreach ( $batch_result as $url_id => $result ) {
+			if ( is_array( $result ) ) {
+				$page_data = array();
+				$page_data[] = $result['page_title'] ?? '';
+
+				$has_h1 = isset( $result['texts'][1][0] ) && 'h1' === $result['texts'][1][0];
+
+				foreach ( $result['texts'] as $element_id => $element ) {
+					if ( ( ! $has_h1 || $element_id > 0 ) ) {
+						foreach ( $element[1] as $line ) {
```

Comment:
we could add to task parameters also opotion to filter by specific tag
e.g. in some cases we want just headers of pages